tests/test_main.py

```diff
@@ -33,5 +36,40 @@ def Shell(*args, **kwargs):
         xonsh.main.premain(['-l'])
         assert_true(builtins.__xonsh_env__.get('XONSH_LOGIN'))
 
+
+def test_login_shell_with_file_argument():
+    with patch('xonsh.main.Shell', Shell), mock_xonsh_env({}):
+        xonsh.main.premain(['tests/sample.xsh'])
+        assert_false(builtins.__xonsh_env__.get('XONSH_INTERACTIVE'))
+
+    for case in ('-i', '-vERSION', '-hAALP','TTTT', '-TT', '--TTT'):
+        with patch('xonsh.main.Shell', Shell), mock_xonsh_env({}):
+            xonsh.main.premain(['tests/sample.xsh', case])
+            assert_false(builtins.__xonsh_env__.get('XONSH_INTERACTIVE'))
+
+
+def test_login_shell_invalid_arguments():
+    # pytest transition
+    # TODO: check for proper error msg in stdout (howto nose?)
+    with patch('xonsh.main.Shell', Shell), mock_xonsh_env({}):
+        try:
+            xonsh.main.premain(['----'])
+            assert False
+        except SystemExit:
+            pass
+    with patch('xonsh.main.Shell', Shell), mock_xonsh_env({}):
+        try:
+            xonsh.main.premain(['--hep'])
+            assert False
+        except SystemExit:
+            pass
+    with patch('xonsh.main.Shell', Shell), mock_xonsh_env({}):
+        try:
+            xonsh.main.premain(['-TT'])
+            assert False
+        except SystemExit:
+           pass
```

Comment:
ah rly? i don't like this so much, i just abuse the fact that argparse throws a SystemExit when it doesn't like something. i tried checking out for error messages on stderr but i think nose messes with that, it will be easier with pytest